### PR TITLE
Add appveyor-powered continuous integration for Windows

### PR DESCRIPTION
**Pull request by lithp**
Continuous integration for windows!

For #1970

### CODE REVIEW

**Comment by lithp** [comment]
It works :boom:

**Comment by jasonmp85** [comment]
It's still failing, though?

**Comment by lithp** [comment]
Well, https://github.com/citusdata/citus/pull/1959 hasn't been merged yet :)

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,118 @@
 # Citus toplevel Makefile
 
+EXTENSION = citus
+DATA = citus--7.0-15.sql
+
+
+MODULE_big = citus
+
+OBJS = src/backend/distributed/shared_library_init.o \
```
Comment (jasonmp85) [inline Makefile]: Hey so this is crazy but what if we generate the Makefile using a minimal Docker image we run on Windows?

File: windows/include/citus_version.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/* Citus full name as a string */
```
Comment (jasonmp85) [inline windows/include/citus_version.h]: Same for this. Maybe we just generate it using a minimal Linux image?

File: Makefile (hunk 1 above)
Comment (lithp) [reply Makefile]: IInteresting. This goes to something I was thinking about yesterday when mulling over `citus_version.h`, I wonder how portable we want this to be. I mean, I can get appveyor running perfectly by installing `autoconf` and `make` and making small changes to our existing makefiles, but I don't know how much help I should be giving other people who might want to build Citus in the future.

Presumably Docker is in the same category. Should we require anyone building Citus on Windows to have docker?

Comment (jasonmp85) [reply Makefile]: I'm not saying to make users install it; maybe we should just check in `.win` versions of these files or something. I'm just saying for the purposes of making a script _we_ can run when we need to to update various files, it might be nice to script it somewhere we're comfortable with tools (i.e. Linux) and just run it against a shared-mount folder when we need to update these things in Windows.

**Comment by pykello** [comment]
Is this in development or waiting for review from someone?

**Comment by lithp** [comment]
@pykello oh, this is waiting on review from you!

**Comment by lithp** [comment]
And also on #2002

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,118 @@
 # Citus toplevel Makefile
```
Comment (pykello) [inline Makefile]: Why is this PR changing this?

File: windows/include/citus_version.h (hunk 2 above)
Comment (pykello) [inline windows/include/citus_version.h]: Why is this PR changing this?

File: src/backend/distributed/utils/statistics_collection.c [hunk 4]
```diff
@@ -25,7 +25,17 @@ PG_FUNCTION_INFO_V1(citus_server_id);
 #ifdef HAVE_LIBCURL
```
Comment (pykello) [inline src/backend/distributed/utils/statistics_collection.c]: Not part of this PR

File: src/backend/distributed/executor/multi_client_executor.c [hunk 5]
```diff
@@ -29,9 +29,8 @@
 #include <errno.h>
```
Comment (pykello) [inline src/backend/distributed/executor/multi_client_executor.c]: Not part of this PR

File: src/backend/distributed/connection/remote_commands.c [hunk 6]
```diff
@@ -995,13 +995,19 @@ BuildWaitEventSet(MultiConnection **allConnections, int totalConnectionCount,
 	WaitEventSet *waitEventSet = NULL;
```
Comment (pykello) [inline src/backend/distributed/connection/remote_commands.c]: Not part of this PR

File: src/backend/distributed/connection/connection_management.c [hunk 7]
```diff
@@ -10,7 +10,10 @@
```
Comment (pykello) [inline src/backend/distributed/connection/connection_management.c]: Not part of this PR

File: Makefile (hunk 3 above)
Comment (lithp) [reply Makefile]: Windows builds don't use Make, they instead use visual studio. There's a script in the postgres repo, Mkvcbuild.pm, which will parse our Makefile and generate the correct visual studio project file. Here we're getting our Makefile into a form which it understands.

Probably a better alternative would be to switch over to CMake, but that's going to be a large project.

File: windows/include/citus_version.h (hunk 2 above)
Comment (lithp) [reply windows/include/citus_version.h]: Similar story. `citus_version.h` is generated by our Makefile. On Windows we don't use Make, so we have to hard-code it. This is very suboptimal but hard to avoid.
